tests/models/test_models.py

```diff
@@ -0,0 +1,225 @@
+"""
+Test model
+"""
+from pathlib import Path
+
+# pylint: disable=too-few-public-methods
+# pylint: disable=no-self-use
+# pylint: disable=missing-class-docstring
+# pylint: disable=unused-argument
+# pylint: disable=abstract-method
+import torch
+
+from nerfstudio.cameras.cameras import Cameras
+from nerfstudio.data.datasets.base_dataset import DataparserOutputs, InputDataset
+from nerfstudio.data.scene_box import SceneBox
+from nerfstudio.models.vanilla_nerf import NeRFModel, VanillaModelConfig
+from nerfstudio.pipelines.base_pipeline import (
+    VanillaDataManagerConfig,
+    VanillaPipeline,
+    VanillaPipelineConfig,
+)
+
+
+class MockedDataManager:
+    """Mocked data manager"""
+
+    def __init__(self, *args, **kwargs):
+        num_images = 0
+        self.train_dataset = InputDataset(
+            DataparserOutputs(
+                image_filenames=[Path("filename.png")] * num_images,
+                cameras=Cameras(
+                    camera_to_worlds=torch.ones([num_images, 3, 4], dtype=torch.float32),
+                    fx=1.0,
+                    fy=1.0,
+                    cx=1.0,
+                    cy=1.0,
+                    width=2,
+                    height=2,
+                ),
+            )
+        )
+
+    def to(self, *args, **kwargs):
+        "Mocked to"
+        return self
+
+
+def test_load_model():
+    """Test that a NeRF Model can be extracted from vanilla Nerf pipeline"""
+
+    # Create dummy vanilla nerf pipeline
+    config = VanillaPipelineConfig(
+        datamanager=VanillaDataManagerConfig(
+            _target=MockedDataManager,
+        ),
+        model=VanillaModelConfig(_target=NeRFModel),
+    )
+    pipeline = VanillaPipeline(config, "cpu")
+
+    state_dict = pipeline.state_dict()
+    # Model weights are set to a known fixed value to compare the result of the model read
+    state_dict[  # pylint: disable=unsubscriptable-object,unsupported-assignment-operation
+        "_model.lpips.net.lin0.model.1.weight"
+    ] = torch.Tensor(
+        [
+            [
+                [[1]],
```

Comment:
can you do list comprehension here? e.g., `[[[1]] for _ in range(N)]` may be cleaner